lib/ansible/plugins/shell/__init__.py

```diff
@@ -154,11 +154,19 @@ def mkdtemp(self, basefile=None, system=False, mode=0o700, tmpdir=None):
         cmd = 'mkdir -p %s echo %s %s' % (self._SHELL_SUB_LEFT, basetmp, self._SHELL_SUB_RIGHT)
         cmd += ' %s echo %s=%s echo %s %s' % (self._SHELL_AND, basefile, self._SHELL_SUB_LEFT, basetmp, self._SHELL_SUB_RIGHT)
 
+        # If basetmp directory or it's parent directories doesn't exist, then we will create it
+        # with 0o77 umask, if it does exist & permission is different then we raise an error
+        # If provide mode is 0o000 then let's default it to 0o700
+        if not mode:
+            mode = 0o700
+        tmp_umask = 0o777 & ~mode
+        if os.path.exists(os.path.realpath(basetmp)):
```

Comment:
this is checking the directory on the controller, not on the remote

Reply:
Can you please elaborate your comment a bit? This is checking the basetmp existence and it's permission after joining it with tmpdir (tmpdir here is system_tmpdirs or remothe_tmp)

Reply:
This existence check executes on the ansbile controller, e.g 'your workstation', while the remote_tmp happens on the target machine, e.g 'webserver'. 

Even if this was being executed on the correct machine you would still have a race condition between check and creation. 

In any case, I already have a code change that fixes this correctly and it will soon be a PR.